scanpipe/tests/test_commands.py

```diff
@@ -595,3 +600,291 @@ def test_scanpipe_management_command_create_user(self):
+                "download_url": "https://registry.npmjs.org/asdf/-/asdf-1.2.2.tgz",
+                "pipelines": ["scan_single_package"],
+            },
+            {
+                "scannable_uri_uuid": scannable_uri_uuid2,
+                "download_url": "https://registry.npmjs.org/asdf/-/asdf-1.2.1.tgz",
+                "pipelines": ["scan_single_package"],
+            },
+        ]
+
+        mock_request_post.side_effect = [
+            {
+                "status": f"updated scannable uri {scannable_uri_uuid1} "
+                "scan_status to failed"
+            },
+            {"status": f"scan indexed for scannable uri {scannable_uri_uuid2}"},
+        ]
+
+        options = [
+            "--max-loops",
+            2,
+        ]
+        out = StringIO()
+        with mock.patch("scanpipe.tasks.execute_pipeline_task", task_failure):
+            call_command("purldb-scan-queue-worker", *options, stdout=out, stderr=out)
+
+        out_value = out.getvalue()
+        self.assertIn(
+            "Project httpsregistrynpmjsorgasdf-asdf-122tgz-97627c6e created", out_value
+        )
+        self.assertIn(
+            "Project httpsregistrynpmjsorgasdf-asdf-121tgz-0bbdcf88 created", out_value
+        )
+        self.assertIn("- asdf-1.2.2.tgz", out_value)
+        self.assertIn("- asdf-1.2.1.tgz", out_value)
+        self.assertIn("Exception occured during scan project:", out_value)
+        self.assertIn("Error during scan_single_package execution:", out_value)
+        self.assertIn("Error log", out_value)
+
+        update_status_url = f"{purldb.PURLDB_API_URL}scan_queue/update_status/"
+        mocked_post_calls = mock_request_post.call_args_list
+        self.assertEqual(2, len(mocked_post_calls))
+
+        mock_post_call1 = mocked_post_calls[0]
+        self.assertEqual(update_status_url, mock_post_call1.kwargs["url"])
+        self.assertEqual(
+            "97627c6e-9acb-43e0-b8df-28bd92f2b7e5",
+            mock_post_call1.kwargs["data"]["scannable_uri_uuid"],
+        )
+        self.assertEqual("failed", mock_post_call1.kwargs["data"]["scan_status"])
+        self.assertIn(
+            "Exception occured during scan project:",
+            mock_post_call1.kwargs["data"]["scan_log"],
+        )
+
+        mock_post_call2 = mocked_post_calls[1]
+        self.assertEqual(update_status_url, mock_post_call2.kwargs["url"])
+        self.assertEqual(
+            "0bbdcf88-ad07-4970-9272-7d5f4c82cc7b",
+            mock_post_call2.kwargs["data"]["scannable_uri_uuid"],
+        )
+        self.assertEqual("failed", mock_post_call1.kwargs["data"]["scan_status"])
+        self.assertIn(
+            "Exception occured during scan project:",
+            mock_post_call2.kwargs["data"]["scan_log"],
+        )
+
+
+class ScanPipeManagementCommandMixinTest(TestCase):
+    class CreateProjectCommand(
+        commands.CreateProjectCommandMixin, commands.AddInputCommandMixin, BaseCommand
+    ):
+        pass
```

Comment:
This setup seems too complex.

Reply:
Do you have any suggestions on how I should test the command methods? Should I have just called `create-project` using `call_command` instead of putting the project creation logic into a method?

Reply:
What about having the functions related to package creation outside the Command class? Those could be at the module level, what's your take?

Reply:
@tdruez I am on board with that idea. What should be done about the messages that are printed using styling added by the django Command class? For instance, the calls to `self.stdout.write()` and `self.stderr.write()`? I think that's the main reason why I kept this code as Command mixins